tools/clang/unittests/HLSLExec/LongVectors.cpp

```diff
@@ -1593,41 +1593,86 @@ template <typename T> T waveMultiPrefixProduct(T A, UINT) {
 
 template <typename T> struct Op<OpType::WaveMatch, T, 1> : StrictValidation {};
 
+// Helper struct to build the expected result for WaveMatch tests.
+struct WaveMatchResultBuilder {
+
+private:
+  uint64_t LowWaveMask;
+  uint64_t HighWaveMask;
+  uint64_t LowBits;
+  uint64_t HighBits;
+
+public:
+  WaveMatchResultBuilder(UINT NumWaves) : LowBits(0), HighBits(0) {
+    const UINT LowWaves = std::min(64U, NumWaves);
+    const UINT HighWaves = NumWaves - LowWaves;
+    LowWaveMask = (LowWaves < 64) ? (1ULL << LowWaves) - 1 : ~0ULL;
+    HighWaveMask = (HighWaves < 64) ? (1ULL << HighWaves) - 1 : ~0ULL;
+    LowBits &= LowWaveMask;
```

Comment:
The assignment to LowBits and HighBits seems redundant? They're initialized to zero so the result of these operations will still always just be 0?